qa/integration-tests/src/test/java/io/zeebe/broker/it/system/BrokerAdminServiceClusterTest.java

```diff
@@ -0,0 +1,153 @@
+import io.zeebe.broker.it.util.GrpcClientRule;
+import io.zeebe.broker.system.management.BrokerAdminService;
+import io.zeebe.broker.system.management.PartitionStatus;
+import io.zeebe.engine.processing.streamprocessor.StreamProcessor.Phase;
+import java.time.Duration;
+import org.awaitility.Awaitility;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.RuleChain;
+import org.junit.rules.Timeout;
+
+public class BrokerAdminServiceClusterTest {
+
+  private final Timeout testTimeout = Timeout.seconds(60);
+  private final ClusteringRule clusteringRule =
+      new ClusteringRule(
+          1,
+          3,
+          3,
+          cfg -> {
+            cfg.getData().setLogIndexDensity(1);
+            cfg.getData().setSnapshotPeriod(Duration.ofMinutes(15));
+          });
+  private final GrpcClientRule clientRule = new GrpcClientRule(clusteringRule);
+
+  @Rule
+  public RuleChain ruleChain =
+      RuleChain.outerRule(testTimeout).around(clusteringRule).around(clientRule);
+
+  private BrokerAdminService leaderAdminService;
+  private Broker leader;
+
+  @Before
+  public void before() {
+    leader = clusteringRule.getBroker(clusteringRule.getLeaderForPartition(1).getNodeId());
+    leaderAdminService = leader.getBrokerAdminService();
+  }
+
+  @Test
+  public void shouldReportPartitionStatusOnFollowersAndLeader() {
+    // given
+    final var followers =
+        clusteringRule.getOtherBrokerObjects(clusteringRule.getLeaderForPartition(1).getNodeId());
+
+    // when
+    final var followerStatus =
+        followers.stream()
+            .map(Broker::getBrokerAdminService)
+            .map(BrokerAdminService::getPartitionStatus)
+            .map(status -> status.get(1));
+
+    final var leaderStatus = leaderAdminService.getPartitionStatus().get(1);
+
+    // then
+    followerStatus.forEach(
+        partitionStatus -> {
+          assertThat(partitionStatus.getRole()).isEqualTo(Role.FOLLOWER);
+          assertThat(partitionStatus.getProcessedPosition()).isNull();
+          assertThat(partitionStatus.getSnapshotId()).isNull();
+          assertThat(partitionStatus.getProcessedPositionInSnapshot()).isNull();
+          assertThat(partitionStatus.getStreamProcessorPhase()).isNull();
+        });
+
+    assertThat(leaderStatus.getRole()).isEqualTo(Role.LEADER);
+    assertThat(leaderStatus.getProcessedPosition()).isEqualTo(-1);
+    assertThat(leaderStatus.getSnapshotId()).isNull();
+    assertThat(leaderStatus.getProcessedPositionInSnapshot()).isNull();
+    assertThat(leaderStatus.getStreamProcessorPhase()).isEqualTo(Phase.PROCESSING);
+  }
+
+  @Test
+  public void shouldReportPartitionStatusWithSnapshotOnFollowers() {
+    // given
+    clientRule.createSingleJob("test");
+    leaderAdminService.takeSnapshot();
+
+    // when
+    waitForSnapshotAtBroker(leaderAdminService);
+
+    // then
+    final var leaderStatus = leaderAdminService.getPartitionStatus().get(1);
```

Comment:
How do we know that 1 is the leader?

Reply:
1 is the partitionId. It is getting the status of partition 1.

Reply:
but it says `leaderStatus`

Reply:
Ya. Status of the leader for partition 1.